Fixed ticket 105 - list flirts for company

SilGosker opened the pull request.

SilGosker committed:
Added types for student and flirts

SilGosker committed:
Changed authentication type

Removed the isAdmin boolean from User, added Role string

SilGosker committed:
Fixed homescreen based on role type.

Added listing flirts for company.

SilGosker committed:
npm run format

SilGosker committed:
Merge branch 'master' into 105-ListFlirtsForCompany

SophiaH67 commented on screens/HomeScreen.tsx:
```diff
@@ -27,14 +28,9 @@ export default function HomeScreen({
   const [user, setUser] = useState<User>();
   const [companies, setCompanies] = useState<User[]>();
-
+  const [flirts, setFlirts] = useState<Flirt[]>();
```
Initialize with empty array instead of undefined (doesn't ts complain about this?

e.g.

```ts
const [myList, setMyList] = useState<string[]>([])
```

HUSKUT commented on screens/HomeScreen.tsx:
```diff
@@ -12,6 +12,7 @@ import TopBar from '../components/TopBar';
 import getLoggedInUser from '../lib/getLoggedInUser';
+import Flirt from '../types/Flirt';
 
 type ProfileScreenNavigationProp = NativeStackNavigationProp<
```
I have fixed this naming issue in my branch

HUSKUT commented on screens/HomeScreen.tsx:
```diff
@@ -44,73 +40,124 @@ export default function HomeScreen({
+      );
+    case 'Moderator':
+      if (!companies) {
+        backendFetch<User[]>('GET', 'company')
```
does this endpoint exist in an upcoming branch/pr?

SophiaH67 commented on screens/HomeScreen.tsx:
```diff
@@ -44,73 +40,124 @@ export default function HomeScreen({
+      );
+    case 'Company':
+      if (!flirts) {
+        backendFetch<Flirt[]>('GET', 'flirts').then(flirts =>
```
Fetches should *always* be in a function that gets called *once*. Such as useEffect or an onClick handler

SophiaH67 commented on screens/HomeScreen.tsx:
```diff
@@ -44,73 +40,124 @@ export default function HomeScreen({
-  );
+          <ScrollView style={styles.content}>
+            {flirts?.map(flirt => (
+              <Notification
+                title={flirt.student.name}
+                key={flirt.id}
+                message={
+                  flirt.student.profileSettings.length > 0
+                    ? flirt.student.profileSettings
+                        .map(
+                          profileSetting =>
+                            profileSetting.key +
+                            ': ' +
+                            profileSetting.value +
+                            '\n'
+                        )
+                        .join('')
+                    : 'Deze student heeft nog niks over zichzelf laten weten'
+                }
+              />
+            ))}
```
Comments + make backend return dictionary instead of array

SophiaH67 commented on types/User.ts:
```diff
@@ -5,7 +5,7 @@ export default interface User {
   name: string;
   email: string;
-  isAdmin?: boolean;
+  role?: 'Moderator' | 'Company' | 'Student';
```
Maybe extract to enum

```ts
enum Role {
  Moderator = 'Moderator'
  Company = 'Company'
  Student = 'Student'
}

export default Role
```

SilGosker committed:
Applied feedback in PR 105

Added enum for role, added profilesetting dictionary

SilGosker committed:
npm run format

SilGosker committed:
Fixed caching bug

Fixed a bug where RN cached the HomeScreen, causing the login to not work correctly

SilGosker committed:
Format fix

HUSKUT committed:
Merge branch 'master' into 105-ListFlirtsForCompany

HUSKUT committed:
fixed an issue where nnavbar still was in code
